src/data/nat/choose.lean

```diff
@@ -100,8 +101,142 @@ theorem add_pow [comm_semiring α] (x y : α) (n : ℕ) :
 (commute.all x y).add_pow n
 
 /-- The sum of entries in a row of Pascal's triangle -/
-theorem sum_range_choose (n : ℕ) : 
+theorem sum_range_choose (n : ℕ) :
   ∑ m in range (n + 1), choose n m = 2 ^ n :=
 by simpa using (add_pow 1 1 n).symm
 
+/-
+Facts about more specific binomial coefficients and their sums
+-/
+
+private lemma succ_not_le {m : nat} (t : m.succ ≤ m) : false :=
+begin
+  induction m, linarith,
+  apply m_ih,
+  exact nat.lt_succ_iff.mp t,
+end
+
+lemma sum_range_split_halfway (f : nat → nat) (m : nat) :
+  ∑ i in range (2 * m + 2), f i = (∑ i in range (m + 1), f i) + ∑ i in (Ico (m + 1) (2 * m + 2)), f i :=
```

Comment:
This should be a general lemma for splitting a sum over the interval (which likely already exists, but I don't know).

Reply:
I generalised it; couldn't find an existing version (which is very poor evidence of it not existing!). Would it go in data/nat/finset?

Reply:
`sum_Ico_consecutive` I guess.